src/Admin/Preview.php

```diff
@@ -21,9 +21,114 @@ function() {
 					$this->registerPreviewStatusFieldsAndMutations();
 				}
 			);
+
+			$use_cloud_loader = self::get_setting( 'use_gatsby_content_sync' );
+			
+			if ( $use_cloud_loader === 'on' ) {
+				add_filter( 'preview_post_link', function( $link, $post ) {
+					return self::get_gatsby_content_sync_url_for_post( $post );
+				}, 10, 2 );
+			}
 		}
 	}
 
+	public static function get_preview_manifest_id_for_post( $post ) {
+		$graphql_single_name = 
```

Comment:
What is a `graphql_single_name`? 

I'm guessing I just don't have the context to what this is actually doing or what this means.

Reply:
When post types are registered in WPGraphQL they get a single name and a plural name https://www.wpgraphql.com/docs/custom-post-types/ 
I'm using the single name as the node type in the `gatsby_action_ref_node_type` taxonomy to make sure we uniquely ID the node (db ID + node type).

Reply:
But also if there's no single name we can't use this node in Gatsby anyway so it returns null if it doesn't exist.